src/WordSearch/WordSearchSolver.ts

```diff
@@ -51,6 +55,7 @@ export class WordSearchSolver {
 
   setGrid(matrix: string[][]) {
     this._matrix = matrix;
+    this.checkSpaces();
```

Comment:
I wonder if we should instead have a "parse" method that takes a single string and have that handle the spaces. Then this can be a pure "I know what I'm doing" call and the "parse method already implies that we will do some sort of processing on the input.

Reply:
Sure